File: backend/src/main/java/com/seedsanskriti/service/ReviewService.java
package com.seedsanskriti.service;

import java.util.List;

import org.springframework.security.core.Authentication;

import com.seedsanskriti.dto.AddReviewRequest;
import com.seedsanskriti.dto.ReviewResponse;

public interface ReviewService {

    ReviewResponse addOrUpdateReview(AddReviewRequest request, Authentication authentication);

    List<ReviewResponse> getProductReviews(Long productId);

    String deleteReview(Long reviewId, Authentication authentication);
}

File: seedsanskriti_frontend/src/services/reviewService.js
import apiClient from './apiClient';

const reviewService = {
  addOrUpdateReview: (payload) => apiClient.post('/reviews', payload).then((res) => res.data),

  getProductReviews: (productId) =>
    apiClient.get(`/reviews/product/${productId}`).then((res) => res.data),

  deleteReview: (reviewId) =>
    apiClient.delete(`/reviews/${reviewId}`).then((res) => res.data),
};

export default reviewService;
